src/core/annotation.js

```diff
@@ -797,8 +797,17 @@ var HighlightAnnotation = (function HighlightAnnotationClosure() {
   function HighlightAnnotation(parameters) {
     Annotation.call(this, parameters);
 
+    var dict = parameters.dict;
     this.data.annotationType = AnnotationType.HIGHLIGHT;
-    this.data.hasPopup = parameters.dict.has('Popup');
+
+    if (!dict.has('C')) {
+      // Fall back to the default background color.
+      this.data.color = null;
+    }
+
+    this.data.hasPopup = dict.has('Popup');
+    this.data.title = stringToPDFString(dict.get('T') || '');
+    this.data.contents = stringToPDFString(dict.get('Contents') || '');
```

Comment:
So, it appears that this code is repeated six times throughout the file. Would it be possible to re-factor into a helper function instead (similar to the other file)?